rviz_common/include/rviz_common/ros_integration/ros_node_abstraction.hpp

```diff
@@ -96,11 +89,18 @@ class RosNodeAbstraction : public RosNodeAbstractionIface
    */
   RVIZ_COMMON_PUBLIC
   std::map<std::string, std::vector<std::string>>
-  get_topic_names_and_types() override;
+  get_topic_names_and_types() const override;
+
+  // TODO(anhosi): remove once the RosNodeAbstraction is extended to handle subscriptions
+  RVIZ_COMMON_PUBLIC
+  rclcpp::Node::SharedPtr
+  get_raw_node() override
+  {
+    return raw_node_;
+  }
 
 private:
-  std::string node_name_;
-  std::shared_ptr<RosNodeStorageIface> ros_node_storage_;
+  rclcpp::Node::SharedPtr raw_node_;
```

Comment:
In my opinion, this is the wrong direction to be moving in for this class. The abstraction class really needs to keep ROS 2 stuff out of it (hence the private node storage class).

I'd rather see a ROS 2 specific subclass of this abstraction interface which can get the rclcpp (ROS 2) node out of the abstraction. That way this class remains abstract but you can introduce another (non ROS 1/ROS 2 agnostic) set of classes to do specific things in the meantime.